### DISCUSSION TITLE
What is the point of DTOs?

### DISCUSSION THREAD

**Discussion by Xemnas0** [litestar-org/litestar#2908] [Q&A] [answered]
I've seen that Litestar implements DTO. However, I don't quite grasp why one would need it in their application.
If I already specify, for example, the request and response schemas with a Pydantic BaseModel, why would I need another layer on top of that?

**Comment by peterschutt** [comment, marked as answer]
Short answer: you don't need them and  there is no requirement for you to use them.

Longer answer is that DTOs can save a lot of repetition by removing the need to define explicit request and response schemas.

> why would I need another layer on top of that?

You wouldn't - they are an alternative to defining explicit request and response schemas.
